modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java

```diff
@@ -278,10 +278,16 @@ public String getTypeDeclaration(Schema p) {
         if (ModelUtils.isArraySchema(p)) {
             ArraySchema ap = (ArraySchema) p;
             Schema inner = ap.getItems();
-            return "[]" + getTypeDeclaration(inner);
+            if (ModelUtils.isGenerateAliasAsModel())
```

Comment:
@zippolyte I don't think you need `ModelUtils.isGenerateAliasAsModel()` as `unaliasSchema` has already taken that into account:  https://github.com/OpenAPITools/openapi-generator/blob/10627ed5cc3f4703b0f02662ea568353e8d26e00/modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java#L812